lib/backends/sdc/build.js

```diff
@@ -891,8 +894,266 @@ function runBuildCommand(opts, callback) {
+    var scratchImage;
+    var socketOpts;
+
+    // There is the potential to run the callback twice, so make sure that never
+    // happens.
+    callback = once(callback);
+
+    try {
+        rat = drc.parseRepoAndTag(opts.tag || '');
+    } catch (e) {
+        callback(new errors.DockerError(e, e.message));
+        return;
+    }
+
+    vasync.pipeline({ funcs: [
+        getScratchImage,
+        findContainerImage,
+        startCnapiDockerCommit,
+        runCnapiDockerCommit
+    ]}, commitCleanup);
+
+
+    function getScratchImage(_, cb) {
+        req.app.backend.getScratchImage(req,
+        function _getScratchImageCb(err, sImg) {
+            if (err) {
+                log.error(err, 'getScratchImage failure');
+                cb(err);
+                return;
+            }
+            scratchImage = sImg;
+            //log.debug('scratchImage: ', scratchImage);
+            cb();
+        });
+    }
+
+    // Find the Image model for running container.
+    function findContainerImage(_, cb) {
+        var dockerImageOpts = {
+            account: req.account,
+            app: req.app,
+            log: log
+        };
+        images.getDockerImageForUuid(vm.image_uuid, dockerImageOpts,
+            function _dockerImageUuidCb(err, imgModel) {
+                baseImg = imgModel;
+                cb(err);
+            }
+        );
+    }
+
+    // Start the cn-agent commit task.
+    function startCnapiDockerCommit(_, cb) {
+        var cnapi = req.app.cnapi;
+        var cnapiCommitPayload = common.objCopy(opts);
+        var cnapiCommitOpts = {
+            headers: {
+                'x-request-id': req_id,
+                // Add the docker registry headers.
+                'x-registry-config': req.headers['x-registry-config'],
+                'x-registry-auth': req.headers['x-registry-auth']
+            }
+        };
+
+        cnapiCommitPayload.account_uuid = req.account.uuid;
+        cnapiCommitPayload.command = 'commit';  // Command type.
+        cnapiCommitPayload.fromImage = utils.imgobjToInspect(baseImg);
+        cnapiCommitPayload.fromImageUuid = vm.image_uuid;
+        cnapiCommitPayload.imgapi_url = req.app.config.imgapi.url;
+
+        /* CNAPI, commit the container to an image for us. */
+        cnapi.dockerBuild(vm.server_uuid, vm.uuid,
+            { payload: cnapiCommitPayload },
+            cnapiCommitOpts,
+            function _dockerCommitCb(err, result) {
+                if (err) {
+                    log.error(err, 'error calling cnapi.dockerCommit');
+                    cb(err);
+                    return;
+                }
+
+                socketOpts = result;
+                cb();
+            }
+        );
+    }
+
+
+    function runCnapiDockerCommit(_, cb) {
+        var commitError;
+        var commitEventStream;
+        var host = socketOpts.host;
+        var port = socketOpts.port;
+
+        var commitSocket = net.createConnection({ host: host, port: port });
```

Comment:
It follows the usual pattern of:

```
commitSocket.on('error', function onError(err) {});
```
and:
````
commitSocket.on('timeout', function onTimeout(){});
````

Note that in the case of the 'timeout' event the socket itself is not closed or modified in any way, so if the desired behavior is to close it, then it'll need to be done manually.